### ISSUE TITLE
pdf's not being converted to img's anymore

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/louisville-hyku#75]
# Summary
refer to [this thread](https://gitlab.com/notch8/louisville-hyku/-/issues/10#note_945156568).

pdf's appear to no longer be getting converted into jpg derivatives

# Acceptance Criteria

* [ ] an imported parent work with a pdf attached as the file, should have jpg derivatives created for each page of the pdf
* [ ] the above jpg's ought to show in the uv viewer for the parent work

# Demo
Url: https://lv-hyku-staging.notch8.cloud/importers/48?locale=en
- the above importer was imported successfully, but no files were attached

# Testing Instructions
*

# Notes
## Engineering Notes

**Comment by orangewolf** [comment]
Related Merge Requests
---
[116-parent-pdf](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/106)

**Comment by orangewolf** [comment]
Alisha May 2022
---
[@rachel.howard](https://github.com/rachel.howard) this is actually not completely a bug. uploading a parent with a pdf vs children with their own jpg's will not give you the same end result. (allowing the uv to still display correctly even when children don't have images, but the parent does, will be deployed to staging tonight)
the csv you were uploading had a "files" header. I uploaded this csv ([warriors_pdf.csv](https://gitlab.com/notch8/louisville-hyku/uploads/26b7e62df421c48d9a00426adbb1eb6f/warriors_pdf.csv)) with a "file" column locally instead. (I did not use the cardinal text csv so I could bypass the ocr reading) the parent work in my new csv referenced a pdf with 3 images in the "file" column. no other works referenced a file at all. the pdf was converted into jpg derivatives which attached to the parent work.
however, the child works do not have images associated with them. this is expected because they have no "file" reference in the csv. the jpgs that you see on the parent work are created on the fly from the pdf which is why they exist on the parent. (they'd all exist on any parent/child work that listed the pdf in its "file" column) this is the approved behavior per [#45 (closed)](https://github.com/notch8/louisville-hyku/-/issues/45).

**Comment by orangewolf** [comment]
Rachel May 2022
---
Ah, ok! This sample set is so old; at the time I didn't know whether we'd go with PDF or JPG files for ingest, and as we've made decisions and modified metadata structure I have just glossed over the fact that those records remained in the CSV. I will test a PDF with metadata only at the parent (PDF) level.

Thanks,

**Comment by orangewolf** [comment]
Alisha May 2022
---
what do you mean a pdf with metadata only? I edited the csv in the zip you sent me to have a "file" header instead of "files". I then uploaded it to our staging server so I could use the server path option. (/app/samvera/hyrax-webapp/tmp/originals/Cardinal_Text/Cardinal_Text_HasParentMetadata_updated.csv)

I ran [this importer](https://lv-hyku-staging.notch8.cloud/importers/49?locale=en) last night after !106 was deployed to staging and it's finished as I expected.*
- the parent work has the children works attached, the pdf, and the jpg derivatives created from the pdf (will have to scroll through all the "items" pages to see them all)
- the parent uv shows the jpg thumbnails
- the child works have no images attached to them
- the child works do have the appropriate metadata

* I had to run the importer twice since the first time is the equivalent of "update metadata". the second time through I was able to choose "update metadata and replace files", which allowed the jpg derivatives to be created


![Image](https://user-images.githubusercontent.com/1054448/204000723-87b58a36-fdf3-4fcc-887e-45e2b2b04d25.png)




| parent uv | child works/file sets |
| ------ | ------ |
| ![Image](https://user-images.githubusercontent.com/1054448/204000774-9c591a07-69f8-452a-8bc1-b4f67dee6aad.png)| 
![Image](https://user-images.githubusercontent.com/1054448/204000855-45d75b68-a732-4663-a993-d11f75f6c756.png)|
| ![Image](https://user-images.githubusercontent.com/1054448/204000978-4436354b-0472-4379-9e21-2a391bcb903a.png) | ![Image](https://user-images.githubusercontent.com/1054448/204001051-65c1561c-1348-4099-878e-5d5f2b23a473.png) |

**Comment by orangewolf** [comment]
Rachel May 2022
---
I meant metadata only at the PDF level - so essentially it's not a parent/child relationship, it's one metadata record per associated PDF.

**Comment by orangewolf** [comment]
Rachel May 2022
---
ref !106ref [!106 (merged)](https://github.com/notch8/louisville-hyku/-/merge_requests/106)
# expected behavior
- still create the manifest if the parent work has images attached but the child works do not

# demo
| | |
| --- | --- |
| parent uv | ![Image](https://user-images.githubusercontent.com/1054448/204001324-6ea8e8bc-3d7a-459c-80e0-ee89c7be98ec.png)|
| parent children |![Image](https://user-images.githubusercontent.com/1054448/204001413-8e894732-9718-4e81-bb83-f1555f116dee.png) |
| cell | cell |

**Comment by orangewolf** [comment]
Randy May 2022
---
@alishaevn Sorry, I accidentally closed this. Shot in the dark... Could it be that this is in any way related to [#109 (closed)](https://gitlab.com/notch8/louisville-hyku/-/issues/109)?[@alishaevn](https://github.com/alishaevn) Sorry, I accidentally closed this. Shot in the dark... Could it be that this is in any way related to [#109 (closed)](https://gitlab.com/notch8/louisville-hyku/-/issues/109)?

**Comment by orangewolf** [comment]
Alisha May 2022
---
it shouldn't be. issue #109 referenced a jpg being turned into a pdf. whereas this ticket is for pdf's being converted to a jpg. they are two different jobs in the code. :slight_smile:

the pdf to jpg job does appear to still be functioning correctly either way.it shouldn't be. issue [#109 (closed)](https://github.com/notch8/louisville-hyku/-/issues/109) referenced a jpg being turned into a pdf. whereas this ticket is for pdf's being converted to a jpg. they are two different jobs in the code. 🙂
the pdf to jpg job does appear to still be functioning correctly either way.

**Comment by orangewolf** [comment]
Rachel May 2022
---
I uploaded Burt a few days ago to our Dev server - it is PDFs; the only parent is the collection. (See attached CSV). Same issue - no thumbnails, no images in the viewer.[Burt_PDF.csv](/uploads/40c73ad487c1fac135ff61e79a2da329/Burt_PDF.csv)

**Comment by orangewolf** [comment]
Alisha May 2022
---
I'm testing this on staging, but I also pushed new code to randy yesterday. can you try this on your dev server again?

**Comment by orangewolf** [comment]
Rachel May 2022
---
Even my Burt PDF zip file was too large so I created a folder in DropBox to share it and one of the Poe parents/JPEG children. It was shared with @alishaevn's scientist.com email - let me know if it doesn't work.Even my Burt PDF zip file was too large so I created a folder in DropBox to share it and one of the Poe parents/JPEG children. It was shared with [@alishaevn](https://github.com/alishaevn)'s scientist.com email - let me know if it doesn't work.

**Comment by orangewolf** [comment]
Alisha May 2022
---
thanks! I noticed that in the "poe" zip, there was a "files_subset" folder. you may have changed that as a reference for me, but if not, that wouldn't work for importing. the folder has to be called "files" only.

I've uploaded both files to our [shared google drive](https://drive.google.com/drive/u/4/folders/1gYgKBpzyVRkOXp8TcZ7q2Z7NVjy5OhCQ) as well.thanks! I noticed that in the "poe" zip, there was a "files_subset" folder. you may have changed that as a reference for me, but if not, that wouldn't work for importing. the folder has to be called "files" only.
I've uploaded both files to our [shared google drive](https://drive.google.com/drive/u/4/folders/1gYgKBpzyVRkOXp8TcZ7q2Z7NVjy5OhCQ) as well.

**Comment by orangewolf** [comment]
Rachel May 2022
---
Yeah, I had to change it because I was creating the zip in the same directory that had the "real" (complete) files folder for this collection. Was hoping you could just rename it.

**Comment by orangewolf** [comment]
Alisha May 2022
---
I successfully uploaded the "poe" csv and files. the parent work shows the images in the uv and each child work has a thumbnail. as does the parent work.

- importer: https://lv-hyku-staging.notch8.cloud/importers/53?locale=en
- parent work: https://lv-hyku-staging.notch8.cloud/concern/texts/f64d5934-8fcb-4c76-9e0c-435cbff25759?locale=en



![Image](https://user-images.githubusercontent.com/1054448/204001950-5e67280c-b692-4885-aa81-96ca8579ff0f.png)

**Comment by orangewolf** [comment]
Alisha May 2022
---
I'm now running the luma burt csv/files. it looks rather large though so it might take a while. if you're still having issues, could you or randy re-run the migrations (see below) and then restart the dev server?

``` bash
# should be one of the following commands
rails db:migrate
bundle exec rails db:migrate
```

**Comment by orangewolf** [comment]
Rachel May 2022
---
Since @rtkuehn did import Poe should he also run the above commands or were they just for if Poe didn't import on our dev server?Since [@rtkuehn](https://github.com/rtkuehn) did import Poe should he also run the above commands or were they just for if Poe didn't import on our dev server?

**Comment by orangewolf** [comment]
Alisha May 2022
---
glad to see that poe imported successfully on the dev server as well. :tada: I responded to randy's comment on #117 though.glad to see that poe imported successfully on the dev server as well. 🎉 I responded to randy's comment on [#117 (closed)](https://github.com/notch8/louisville-hyku/-/issues/117) though.

**Comment by orangewolf** [comment]
Alisha May 2022
---
moving this to client verification since the poe import is working on staging and on the lv dev server

**Comment by orangewolf** [comment]
Rachel May 2022
---
We had a lot going on in this ticket but I just looked at Burt on staging (since that one is actually PDFs; Poe was JPGs) and it did convert to images, yay! However, OCR does not seem to be working on Burt (see #117).We had a lot going on in this ticket but I just looked at Burt on staging (since that one is actually PDFs; Poe was JPGs) and it did convert to images, yay! However, OCR does not seem to be working on Burt (see [#117 (closed)](https://github.com/notch8/louisville-hyku/-/issues/117)).
